src/it/scala/com.rbmhtechnology.calliope/TransactionalEventProducerSpec.scala

```diff
@@ -229,16 +231,34 @@ class TransactionalEventProducerSpec extends KafkaSpec with MustMatchers with Ty
       "delete produced events from the underlying event store" in {
         val (topic, consumer) = topicConsumer()
         val eventStore = WritableEventStore()
-        val writer = runTransactionalEventProducer(topic, eventStore, settings.withDeleteInterval(500.millis))
+        val deleteInterval = 500.millis
+        val writer = runTransactionalEventProducer(topic, eventStore, settings.withReadBufferSize(10).withDeleteInterval(deleteInterval))
 
         writer.writeEvent(Event("1", "agg1"))
         writer.writeEvent(Event("2", "agg1"))
         writer.writeEvent(Event("3", "agg1"))
 
-        consumer.request(3)
-        consumer.expectNextN(3)
+        Thread.sleep((deleteInterval * 2).toMillis)
```

Comment:
It seems we need pretty huge delays between these steps. I tried with factor 1.5 and it failed. Is it clear to you why we need such a long delay?